vm/vm.go

```diff
@@ -133,12 +161,39 @@ type VM struct {
 	stop  chan struct{}
 }
 
-func New(factory ControllerFactory, v *version.Semantic) *VM {
-	return &VM{
-		factory: factory,
-		v:       v,
-		config:  NewConfig(),
+func New(
+	factory ControllerFactory,
+	v *version.Semantic,
+	actionRegistry chain.ActionRegistry,
+	authRegistry chain.AuthRegistry,
+	authEngine map[uint8]AuthEngine,
+	opts ...Option,
+) (*VM, error) {
+	vm := &VM{
+		factory:        factory,
+		v:              v,
+		config:         NewConfig(),
+		actionRegistry: actionRegistry,
+		authRegistry:   authRegistry,
+		authEngine:     authEngine,
+	}
+
+	txGossiper, err := gossiper.NewProposer(vm, gossiper.DefaultProposerConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	// Set defaults
+	vm.options = &Options{
+		BlockBuilder: builder.NewTime(vm),
+		TxGossiper:   txGossiper,
 	}
+
+	for _, opt := range opts {
+		opt.Apply(vm, vm.options)
+	}
```

Comment:
Thought - Could go either way. What do you think of setting factories here and initializing within `Initialize` ?

I think as long as we're not starting any goroutines or allocating much here it's fine to do in `New` instead, but the way I had thought about it was passing in factories to `New` so that we're doing the same thing in `Initialize`.

Reply:
This would require another interface to implement which we don't currently need... technically we could be more consistent by initializing within `Initialize` but I want to remove `Initialize` from the VM in the long-term